integrationTests/chainSimulator/bridge/executeAndBurn_test.go

```diff
@@ -234,3 +255,47 @@ func getBridgedValue(bridgeInTokens []chainSim.ArgsDepositToken, token string) (
 	}
 	return nil, fmt.Errorf("token not found")
 }
+
+func requireMetadataInAccount(
+	t *testing.T,
+	nodeHandler process.NodeHandler,
+	token chainSim.ArgsDepositToken,
+	account string,
+	expectedAmount *big.Int,
+) {
+	address := esdtSystemAccount
+	if token.Type == core.Fungible || token.Type == core.NonFungibleV2 {
+		address = account
+	}
+
+	accountKeys, _, err := nodeHandler.GetFacadeHandler().GetKeyValuePairs(address, dataApi.AccountQueryOptions{})
```

Comment:
Wouldn't it be easier to call the api endpoint to return all tokens?
Just asking, nothing to be changed if not needed.

Reply:
I'm using the api endpoint with `RequireAccountHasToken` after all operations.
Metadata verification is used because:
1. I'm checking the `token type`
2. depending on ESDT type, the metadata should be in user account or system account